pkg/shell/active-pages-modal.jsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2021 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+
+import React from "react";
+import { ListingTable } from "cockpit-components-table.jsx";
+import { Button, Label, Split, SplitItem, Modal } from '@patternfly/react-core';
+
+const _ = cockpit.gettext;
+
+export class ActivePagesDialog extends React.Component {
+    constructor(props) {
+        super(props);
+
+        const frames = [];
+        for (const address in props.frames.iframes) {
+            for (const component in props.frames.iframes[address]) {
+                const iframe = props.frames.iframes[address][component];
+                frames.push({
+                    frame: iframe,
+                    component: component,
+                    address: address,
+                    name: iframe.getAttribute("name"),
+                    visible: iframe.getAttribute("data-active") === 'true',
+                    selected: iframe.getAttribute("data-active") === 'true',
+                    displayName: address === "localhost" ? "/" + component : address + ":/" + component
+                });
+            }
+        }
+
+        // sort the frames by displayName, visible ones first
+        frames.sort(function(a, b) {
+            return (a.visible ? -2 : 0) + (b.visible ? 2 : 0) +
+                   ((a.displayName < b.displayName) ? -1 : 0) + ((b.displayName < a.displayName) ? 1 : 0);
+        });
+
+        this.state = { frames: frames };
+
+        this.onRemove = this.onRemove.bind(this);
+    }
+
+    onRemove() {
+        this.state.frames.forEach(element => {
+            if (element.selected)
+                this.props.frames.remove(element.host, element.component);
+        });
+        this.props.onClose();
+    }
+
+    render() {
+        const self = this;
+        const frames = self.state.frames.map(frame => {
+            const columns = [{
+                title: <Split>
+                    <SplitItem isFilled>
+                        {frame.displayName}
+                    </SplitItem>
+                    <SplitItem>
+                        {frame.visible && <Label color="blue">{_("active")}</Label>}
```

Comment:
So this maps `data-active` to frame.visible and back to label "active". Could this be called frame.active then to reduce confusion?

Reply:
yes, will do